lib/room.cpp

```diff
@@ -400,46 +458,33 @@ class Room::Private {
                                        const QString& eventId,
                                        QDateTime timestamp)
     {
-        std::pair<QString, uint32_t> decrypted;
-        // new e2ee TODO:
-        /*
         QPair<QString, QString> senderSessionPairKey =
             qMakePair(senderKey, sessionId);
-        if (!groupSessions.contains(senderSessionPairKey)) {
-            qCDebug(E2EE) << "Unable to decrypt event" << eventId
+        if (groupSessions.find(senderSessionPairKey) == groupSessions.end()) {
+            qCWarning(E2EE) << "Unable to decrypt event" << eventId
                           << "The sender's device has not sent us the keys for "
                              "this message";
             return QString();
         }
-        QOlmInboundGroupSession* senderSession =
-            groupSessions.value(senderSessionPairKey);
-        if (!senderSession) {
-            qCDebug(E2EE) << "Unable to decrypt event" << eventId
-                          << "senderSessionPairKey:" << senderSessionPairKey;
-            return QString();
-        }
-        try {
-            decrypted = senderSession->decrypt(cipher);
-        } catch (QOlmError* e) {
-            qCDebug(E2EE) << "Unable to decrypt event" << eventId
-                          << "with matching megolm session:" << e->what();
+        auto& senderSession = groupSessions[senderSessionPairKey];
+        auto decryptResult = senderSession->decrypt(cipher);
+        if(std::holds_alternative<QOlmError>(decryptResult)) {
+            qCWarning(E2EE) << "Unable to decrypt event" << eventId
+            << "with matching megolm session:" << std::get<QOlmError>(decryptResult);
             return QString();
         }
-        QPair<QString, QDateTime> properties = groupSessionIndexRecord.value(
-            qMakePair(senderSession->id(), decrypted.second));
+        std::pair<QString, uint32_t> decrypted = std::get<std::pair<QString, uint32_t>>(decryptResult);
+        QPair<QString, QDateTime> properties = groupSessionIndexRecord.value(qMakePair(senderSession->sessionId(), decrypted.second));
```

Comment:
```suggestion
        const auto properties = groupSessionIndexRecord.value({senderSession->sessionId(), decrypted.second});
```
See also above about structured bindings (you might need to use `std::pair` for that, didn't check).